transformer/node_dispatcher.ts

```diff
@@ -1,29 +1,20 @@
 import * as ts from 'typescript';
-import { wrapAllPipeableOperators, createWrapCreationExpression, wrapSubscribeMethod } from './operator_wrapper';
+import { createWrapCreationExpression, wrapSubscribeMethod, wrapPipeStatement } from './operator_wrapper';
 
 const rxjsCreationOperators = ['ajax', 'bindCallback', 'bindNodeCallback', 'defer', 'empty', 'from', 'fromEvent',
     'fromEventPattern', 'generate', 'interval', 'of', 'range', 'throwError', 'timer', 'iif'];
 
-type NodeType = 'UNCLASSIFIED' | 'RXJS_CREATION_OPERATOR' | 'RXJS_JOIN_CREATION_OPERATOR' | 'RXJS_PIPE_OPERATOR' | 'RXJS_SUBSCRIBE';
+type NodeType = 'UNCLASSIFIED' | 'RXJS_CREATION_OPERATOR' | 'RXJS_JOIN_CREATION_OPERATOR' | 'RXJS_PIPE' | 'RXJS_SUBSCRIBE';
 
 // Determine if given node is RxJS Creation Operator Statement.
-const isRxJSCreationOperator = (node: ts.Node): [boolean, string] => {
-    try {
-        if (!ts.isCallExpression(node)) {
-            return [false, null];
-        }
-
-        const operator = rxjsCreationOperators
-            .filter(operator => operator === node.expression.getText())
-            .pop();
-
-        return operator ? [true, operator] : [false, null];
-    } catch (e) {
-        // console.log(e);
-        return [false, null];
+const isRxJSCreationOperator = (node: ts.Node): boolean => {
+    if (ts.isCallExpression(node) && ts.isIdentifier(node.expression) && node.expression.getSourceFile() !== undefined) {
+        return rxjsCreationOperators.some(operator => operator === node.expression.getText());
     }
+    return false;
 };
 
+// TODO: cleanup like isRxJSCreationOperator.
 // Determine if given node is given method call.
 const isMethodCall = (node: ts.Node, method: string): boolean => {
     try {
```

Comment:
`        return result.length ? true : false;`
this could be simplyfied. Result must be a list / array type so it should have the length property of Number type
i would suggest:
`        return result.length  > 0;`